webapp/metrics/helper.py

```diff
@@ -37,12 +37,15 @@ def build_metrics_json(
     """
     end = get_last_metrics_processed_date()
 
-    # -1 day counteracts an issue that the api call is inclusive of the dates
-    # specified, meaning you receive 1 extra data point then required
     if metric_bucket == "d":
-        start = end - relativedelta.relativedelta(days=metric_period)
+        start = end + relativedelta.relativedelta(days=-metric_period)
     elif metric_bucket == "m":
-        start = end - relativedelta.relativedelta(months=metric_period)
+        start = end + relativedelta.relativedelta(months=-metric_period)
+    elif metric_bucket == "y":
+        # Go back an extra day to ensure the granularity increases
+        start = end + relativedelta.relativedelta(
+            years=-metric_period, days=-1
```

Comment:
no -1 or : `{'error_list': [{'message': 'Filter interval larger than 365 days at index 0', 'code': 'invalid-date-interval', 'extra': {'index': 0, 'end': '2019-01-28', 'start': '2018-01-27', 'max_days': 365}}]}`

Reply:
The extra day ensures the granularity decreases. Either way, removing the -1 day will still cause an error for a 2 year window. I'll discuss with @cprov why this limit is in place and where it came from, as this worked last week.